pkg/download/download_test.go

```diff
@@ -39,7 +39,7 @@ func TestGetConfigs(t *testing.T) {
 	fileManager := util.CreateTestFileSystem()
 	envs["e1"] = env
 	err := getConfigs(fileManager, "", envs, "")
-	assert.NilError(t, err)
+	assert.ErrorContains(t, err, "There were some errors")
```

Comment:
Why is test changed to assume an error? This looks like a basic sunny case :/

Reply:
I thought the same! Before, the errors were simply not returned. 

The error that occurs is also a funny one IMO. Because Monaco actually tries to query the configs from `https://test.live.dynatrace.com`, fails because it is an unknown tenant, and now also returns the error. 
I think we have to re-engineer quite some parts in Monaco to make it properly testable with mock HTTP clients.

Reply:
Okay, I guess it would make sense to start on that with download in v2.

The dep inversion/injection pattern you've already started with setup in a public method and injection into the private one seems to work quite nicely and is a non-invasive change - you can refactor method per method keeping the public interface unchanged while enabling testing :)